src/rust-crypto/rust-crypto.ts

```diff
@@ -556,7 +567,45 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
             await this.secretStorage.store("m.cross_signing.master", crossSigningPrivateKeys.masterKey);
             await this.secretStorage.store("m.cross_signing.user_signing", crossSigningPrivateKeys.userSigningKey);
             await this.secretStorage.store("m.cross_signing.self_signing", crossSigningPrivateKeys.self_signing_key);
+
+            if (setupNewKeyBackup) {
+                await this.resetKeyBackup();
+            }
```

Comment:
this seems to be adding new functionality, which doesn't really fit with the title of this PR (which implies we are just replacing the existing implementation). Maybe we should separate the new rust impl from the changes to the existing code? It's hard to review at the moment as it feels like we're doing two quite different things.

Reply:
I think that's the main point of the PR? to have boostraping complete now and creating a complete backup?

Reply:
> I think that's the main point of the PR? to have boostraping complete now and creating a complete backup?

Yeah, that's probably fair. But then we should rename the PR to reflect what it's actually doing.

The problem is, I'm struggling to come up with a short summary of the PR. "Add new `CryptoApi.resetKeyBackup` and `CryptoApi.deleteKeyBackupVersion` APIs, and enable backups in `RustCrypto.bootstrapSecretStorage`". What a mouthful!

All of which is to say... I feel like this PR is doing lots of things, and we should probably have broken it down. Which may be one of the reasons it has had 119 comments so far, and multiple rounds of review.

It may be too late now, but I think we can learn from this in the future.

>